sdks/python/apache_beam/testing/benchmarks/nexmark/nexmark_launcher.py

```
@@ -0,0 +1,199 @@
+  assert topic.exists()
+  assert sub.exists()
+
+  logging.info('Generating %d auction events to topic %s',
+               EVENT_COUNT_PER_TYPE*3, topic.name)
+  for _ in xrange(EVENT_COUNT_PER_TYPE):
+    topic.publish(PERSON_EVENT)
+    topic.publish(AUCTION_EVENT)
+    topic.publish(BID_EVENT)
+
+  logging.info('Finished event generation.')
+
+  # Read from PubSub into a PCollection.
+  if args.subscription:
+    raw_events = pipeline | 'ReadPubSub' >> beam.io.ReadFromPubSub(
+        subscription=args.subscription)
+  else:
+    raw_events = pipeline | 'ReadPubSub' >> beam.io.ReadFromPubSub(
+        topic=args.topic)
+
+  return raw_events
+
+
+def run_query(query, args, pipeline_options):
+  try:
+    pipeline = beam.Pipeline(options=pipeline_options)
+    raw_events = generate_events(pipeline, args, pipeline_options)
+    query.load(raw_events)
+    result = pipeline.run()
+    job_duration = pipeline_options.view_as(TestOptions).wait_until_finish_duration # pylint: disable=line-too-long
+    if pipeline_options.view_as(StandardOptions).runner == 'DataflowRunner':
+      result.wait_until_finish(duration=job_duration)
+      result.cancel()
+    else:
+      result.wait_until_finish()
+  except NotImplementedError as exp:
+    print(exp)
+
+
+def run(argv=None):
+  args, _ = parse_args()
+
+  queries = {
+      0: query0,
+      # TODO(mariagh): Add more queries
+  }
+
+  for i in args.query:
+    args, pipeline_options = parse_args()
+    logging.info('Running query %d', i)
+
+    job_setup_time = 20 * 1000    # in milliseconds
+    query_duration = pipeline_options.view_as(TestOptions).wait_until_finish_duration + job_setup_time # pylint: disable=line-too-long
+    command = Command(run_query, args=[queries[i], args, pipeline_options])
+    try:
+      command.run(timeout=query_duration/1000)
+    except Exception as exp:
+      logging.error('Command failed with "%s"', exp)
+
+  logging.info('Queries run successfully: %s', args.query)
```

It says 'successfully', but some queries may have failed. This is because above try/except logs and ignores the exceptions.

---

I added handling for this.